### ISSUE TITLE
prod build warning

### ISSUE THREAD

**Issue by adambreznicky** [TNRIS/tnris.org#9]
when running a prod build, command `npm run dist`, the console displays the deprecation warning below.
this is most likely occurring due to a method being used in the gulpfile. the fact that it doesn't appear when running a dev build indicates that there is a good chance it's from a method in a gulp task only being used in the prod build and not the dev build. if not, it may be webpack config related.

```
(node:27323) [DEP0097] DeprecationWarning: Using a domain property in MakeCallback is deprecated. Use the async_context variant of MakeCallback or the AsyncResource class instead.
```

investigate so we can get ahead of this deprecation.

**Comment by adambreznicky** [comment]
octogonz is onto something: https://www.bountysource.com/issues/72452391-fix-deprecation-warning

issue is related to terser-webpack-plugin creating a .cache folder an dropping a cache within it inside the node_modules on prod build. on initial prod build, the warning appears and this folder doesn't exist. doing a prod build a 2nd time creates this folder and the warning isn't present. subsequent prod builds won't have the warning again unless you delete the .cache folder from node_modules. then the  warning comes back for the initial prod build afterward, but goes away again for every subsequent prod build after that...

**Comment by adambreznicky** [comment]
latest version of terser-webpack-plugin resolved the issue but webpack 4 doesn't require latest  so had to explicitly add it to package.json to be installed. upon webpack 5 release this explicit install can be removed from the package.json.

global install of gulp conflicted with this explicit install also. global gulp is no longer necessary so it has been removed as a requirement. updated jenkins configurations to reflect this.

resolved f54557d83e52843a9c4650187a380090897bb0b7 & resolved bc0d5d3f304d897a95ab3ae97019e1a27c31389d
